src/compiler/checker.ts

```diff
@@ -38841,11 +38841,17 @@ namespace ts {
             if (!checkGrammarDecoratorsAndModifiers(node) && hasEffectiveModifiers(node)) {
                 grammarErrorOnFirstToken(node, Diagnostics.An_export_assignment_cannot_have_modifiers);
             }
+
+            const typeTag = isInJSFile(node) ? getJSDocTypeTag(node) : undefined;
+            if (typeTag) {
+                checkAssertionWorker(typeTag.typeExpression.type, typeTag.typeExpression.type, node.expression);
+                return;

```

Comment:
I feel like we should add type checking to both TS and JS, and TS will never have a type annotation. In that case, returning is not correct; instead it should check the type and continue.

I'm not sure how to make sure that the type of `export default` is the annotation type instead of the expression type either.